src/Language/Java/Inline/Magic.hs

```diff
@@ -0,0 +1,74 @@
+-- | Internal module defining some magic, kept separate from the rest, that
+-- depends on compiler internals.
+
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE MagicHash #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE UnboxedTuples #-}
+
+module Language.Java.Inline.Magic
```

Comment:
How about `Language.Java.Inline.DotClass` or `Language.Java.Inline.Magic.DotClass`? `Magic` seems too vague for which magic the module is allowed to have. The module haddock above is equally vague too.